File: Programmers/LV1/같은 숫자는 싫어/같은 숫자는 싫어.py
# 2020/11/25
# 같은숫자는 싫어 프로그래머스 LV1

arr = [1,1,3,3,0,1,1]

def solution(arr):
    answer = []
    for i in range(len(arr)):
        if i ==0:
            answer.append(arr[i])
        elif arr[i] != arr[i-1]:
            answer.append(arr[i])
    return answer

print(solution(arr))
#set => 중복제거되지만 순서는 오름차순됨

File: Programmers/LV1/JAVA/같은숫자는싫어.java
package Programmers.level1;

import java.util.ArrayList;

public class 같은숫자는싫어 {

	public static void main(String[] args) {

		ArrayList<Integer> list = new ArrayList<Integer>();

		int[] arr = { 1, 1, 3, 3, 0, 1, 1 };

		int temp = Integer.MIN_VALUE;


		for (int i = 0; i < arr.length; i++) {
			if(temp != arr[i]) {
				temp = arr[i];
				list.add(temp);
			}
		}

		int[] answer = new int[list.size()];

		for (int i = 0; i < list.size(); i++) {
			answer[i] = list.get(i);
			System.out.println(answer[i] + " ");
		}

	}

}
